File: src/com/mcblox/parkour/utils/Console.java
package com.mcblox.parkour.utils;

import org.bukkit.Bukkit;
import org.bukkit.ChatColor;

public class Console {
    
    public static void info(String s) {
        Bukkit.getConsoleSender().sendMessage(ChatColor.GREEN + "[Parkour] " + s);
    }
    
    public static void warn(String s) {
        Bukkit.getConsoleSender().sendMessage(ChatColor.YELLOW + "[Parkour] " + s);
    }
    
    public static void err(String s) {
        Bukkit.getConsoleSender().sendMessage(ChatColor.RED + "[Parkour] " + s);
    }
    
    public static void consoleSendMessage(String s) {
        Bukkit.getConsoleSender().sendMessage(ChatColor.translateAlternateColorCodes('&', s));
    }

	public static void catchError(Exception e, String loc) {
		// Error exception catch
		err("There was an error in: " + loc);
		consoleSendMessage(e.toString());
		for (StackTraceElement elem : e.getStackTrace()) {
			if (elem.toString().startsWith("com.mcblox.")) {
				consoleSendMessage("\t" + elem.toString());
			}
		}
		err("End of error");
	}
    
}

File: src/org/serversmc/parkour/utils/Console.kt
package org.serversmc.parkour.utils

import org.bukkit.*
import org.serversmc.parkour.enums.*

object Console {
	
	val consoleSender = Bukkit.getConsoleSender()
	private const val p = "[AutoRestart] "
	
	fun info(s: String) = Bukkit.getConsoleSender().sendMessage("${ChatColor.GREEN}$p$s")
	fun warn(s: String) = Bukkit.getConsoleSender().sendMessage("${ChatColor.YELLOW}$p$s")
	fun err(s: String) = Bukkit.getConsoleSender().sendMessage("${ChatColor.RED}$p$s")
	fun consoleSendMessage(s: String) = consoleSender.sendMessage(translateChatColor('&', s))
	
	fun catchError(e: Exception, loc: String) {
		err("There was an error in $loc")
		consoleSendMessage(e.toString())
		e.stackTrace.forEach { if (it.toString().startsWith("org.serversmc")) consoleSendMessage("\t" + it.toString()) }
		err("End of error")
	}
	
}
